source/extensions/filters/network/redis_proxy/router_impl.h

```diff
@@ -23,28 +25,59 @@ namespace RedisProxy {
 
 typedef std::map<std::string, ConnPool::InstanceSharedPtr> Upstreams;
 
+class MirrorPolicyImpl : public MirrorPolicy {
+public:
+  MirrorPolicyImpl(const envoy::config::filter::network::redis_proxy::v2::RedisProxy::PrefixRoutes::
+                       Route::RequestMirrorPolicy&,
+                   const ConnPool::InstanceSharedPtr, Runtime::Loader& runtime);
+  ~MirrorPolicyImpl() = default;
+
+  ConnPool::InstanceSharedPtr upstream() const override { return upstream_; };
+
+  bool shouldMirror(const std::string& command) const override;
+
+private:
+  const std::string runtime_key_;
+  const envoy::type::FractionalPercent default_value_;
+  const bool exclude_read_commands_;
+  ConnPool::InstanceSharedPtr upstream_;
+  Runtime::Loader& runtime_;
+};
+
+class Prefix : public Route {
+public:
+  Prefix(
+      const envoy::config::filter::network::redis_proxy::v2::RedisProxy::PrefixRoutes::Route route,
+      Upstreams& upstreams, Runtime::Loader& runtime);
+
+  ConnPool::InstanceSharedPtr upstream() const override { return upstream_; }
+  const MirrorPolicies& mirrorPolicies() const override { return mirrorPolicies_; };
+  const std::string& prefix() const { return prefix_; }
+  bool removePrefix() const { return remove_prefix_; }
+
+private:
+  const std::string prefix_;
+  const bool remove_prefix_;
+  const ConnPool::InstanceSharedPtr upstream_;
+  MirrorPolicies mirrorPolicies_;
```

Comment:
`mirror_policies_` (I've noticed a few other non-snake case variables in the redis code, please check)